src/main/java/org/apache/maven/plugin/compiler/CompilationFailureException.java

```diff
@@ -43,13 +43,11 @@ public static String longMessage( List<CompilerMessage> messages )
     {
         StringBuilder sb = new StringBuilder();
 
-        if ( messages != null )
+        for ( CompilerMessage compilerError : messages )
```

Comment:
Let's use this opportunity to improve javadoc as well and to make clear that messages cannot be `null`.

Reply:
Sounds good to me - how's it looking now?

Reply:
Should these methods be `private static` or just inlined into the constructor? They have no other callers

Reply:
Oh, it's not possible to inline them because of the `super()` call